[build] fix for Xamarin.Android.Build.Tasks build

jonathanpeppers opened the pull request:
One issue here is that `obj/Debug/Profile.g.cs` is getting regenerated
on every build, forcing `Xamarin.Android.Build.Tasks.csproj` to rebuild
*every* time. It looks like the `IncrementalClean` target was the culprit,
in that it deleted this file after the build completed.

A simple fix here is to move the file up one directory to
`$(BaseIntermediateOutputPath)`, but not sure if this is the *right*
way to be fixing this.

dnfclas commented:
@jonathanpeppers,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by .NET Foundation. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

dellis1972 commented:
We should be more msbuild friendly and add the Profile.g.cs to the `@(FileWrites)` item group. Infact any target which generates code should do that.

dellis1972 commented:
That said, `IncrementalClean` is a weird beast.. I've had items in the `@(FileWrites)` before and they still get removed.. but its worth a try.

jonathanpeppers commented:
I tried a couple things here, which didn't work 😞 

Added the item to `FileWrites`:
```
    <ItemGroup>
      <FileWrites Include="$(_GeneratedProfileClass)" />
    </ItemGroup>
```

Renamed this target to `BeforeBuild` to run it earlier.

For some reason `IncrementalClean` keeps deleting this file:

![image](https://user-images.githubusercontent.com/840039/32376657-687aaad2-c073-11e7-88ef-b24aafd5fe62.png)

jonathanpeppers commented:
Ok, adding to `FileWrites` seems to work if it includes the absolute path to the file.
